### PR TITLE
Only attach a click listener to a follow button once

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## Description

The preview profile e2e tests had been occasionally flaking, when the Follow button click didn't result in the correct visual feedback. It's been pretty difficult to debug this locally as it fails so infrequently. What I've been able to establish is that when the test fails:

- The follow button has the `data-click-initialized` attribute. This means that the JS code has definitely run and attached the click handler
- I was able to see the code for adding the `onclick` event being correctly triggered
- The correct "Follow" button was being targeted by the test steps
- The code within the `onclick` handler was not being executed

We'd previously noticed that this code sets `button.onclick` rather than adding an event listener, and I can see that this is happening _twice_ for every button. This is due to `initializeLocalStorageRender` and `initializePage` both triggering the initialisation of these buttons.

My suspicion is that `onclick` was being used in this code to avoid a situation where we add two event listeners that cancel each other out (one following a user, then the other immediately unfollowing). I've changed this code so we only ever attach one event listener. 

I haven't yet been able to reproduce a test failure since refactoring this, and although I'm not 100% convinced this will definitely fix the flake, it's a step in the right direction as now the `data-click-initialized` attribute definitely means "no further event handlers are going to be added to this button", which wasn't true before.

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

- Run the app locally and check that follow buttons still behave as expected (e.g. follow a user, refresh the page and check the follow status is persisted, and same for unfollow)
- Run the profile preview spec locally a few times and check it doesn't flake

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [X] This change does not need to be communicated, and this is why not: reinstating an existing test

## [optional] Are there any post deployment tasks we need to perform?

If we merge this and the test starts flaking again we will need to re-skip it and go back to the drawing board

## [optional] What gif best describes this PR or how it makes you feel?

woman with magnifying glass

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeAllFollowButts.js [hunk 1]
```diff
@@ -125,12 +125,14 @@ function fetchButt(butt, buttInfo) {
...
   // currently lacking error handling
   buttons.forEach((butt) => {
-    assignInitialButtResponse(response, butt);
-    butt.onclick = function (e) {
-      e.preventDefault();
-      handleOptimisticButtRender(butt);
-    };
-    butt.dataset.clickInitialized = 'true';
+    if (butt.dataset.clickInitialized !== 'true') {
+      assignInitialButtResponse(response, butt);
```
Comment (citizen428) [inline app/assets/javascripts/initializers/initializeAllFollowButts.js:129]: "Initial Butt Response" 🤣

**Review by citizen428 — APPROVED** [review]
"Initial Butt Response" and "Optimistic Butt Render" 🤣 We really need to rename these at one point.

**Comment by aitchiss** [comment]
> "Initial Butt Response" and "Optimistic Butt Render" 🤣 We really need to rename these at one point.

Nooo we need to never rename these 😆  It brings too much joy

**Review by rhymes — APPROVED** [review]
Let's give this a try 🔥

**Pull request merged**